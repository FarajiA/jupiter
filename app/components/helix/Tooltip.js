import React from 'react';
import PropTypes from 'prop-types';

const Tooltip = (props) => {
  const { id, type, position, children } = props;
  return (
    <div className="tooltip">
      {' '}
      <hx-icon id={`${id}-tooltip`} type={type} />
      <hx-tooltip for={`${id}-tooltip`} position={position}>
        {children}
      </hx-tooltip>
    </div>
  );
};

Tooltip.propTypes = {
  children: PropTypes.node,
  id: PropTypes.string.isRequired,
  position: PropTypes.string,
  type: PropTypes.string
};

Tooltip.defaultProps = {
  position: 'right-middle',
  type: 'help-circle'
};
export default Tooltip;
