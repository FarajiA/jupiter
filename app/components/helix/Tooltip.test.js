import React from 'react';
import renderer from 'react-test-renderer';
import Tooltip from './Tooltip';

describe('Tooltip', () => {
  const props = {
    id: 'testing',
    children: '<h1>Tooltip Content</h1>'
  };

  test('it renders correct props', () => {
    const component = renderer.create(<Tooltip {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders the children', () => {
    const wrapper = shallow(<Tooltip {...props} />).find('hx-tooltip');
    expect(wrapper.text()).toEqual('<h1>Tooltip Content</h1>');
  });
});
