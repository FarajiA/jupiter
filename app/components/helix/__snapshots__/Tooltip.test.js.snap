// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip it renders correct props 1`] = `
<div
  className="tooltip"
>
   
  <hx-icon
    id="testing-tooltip"
    type="help-circle"
  />
  <hx-tooltip
    for="testing-tooltip"
    position="right-middle"
  >
    &lt;h1&gt;Tooltip Content&lt;/h1&gt;
  </hx-tooltip>
</div>
`;
